libnavigation-core/src/main/java/com/mapbox/navigation/core/history/model/HistoryEventMapper.kt

```diff
@@ -0,0 +1,61 @@
+package com.mapbox.navigation.core.history.model
+
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.navigation.core.navigator.toLocation
+import com.mapbox.navigation.navigator.internal.ActiveGuidanceOptionsMapper.mapToActiveGuidanceMode
+import com.mapbox.navigation.navigator.internal.ActiveGuidanceOptionsMapper.mapToGeometriesCriteria
+import com.mapbox.navigator.HistoryRecord
+import com.mapbox.navigator.HistoryRecordType
+import com.mapbox.navigator.Waypoint
+
+internal class HistoryEventMapper {
+
+    fun map(historyRecord: HistoryRecord): HistoryEvent {
+        return when (historyRecord.type) {
+            HistoryRecordType.UPDATE_LOCATION -> mapUpdateLocation(historyRecord)
+            HistoryRecordType.GET_STATUS -> mapGetStatus(historyRecord)
```

Comment:
What's the use for exposing the `GET_STATUS` data? It's not needed for replaying updates, is it?

Given the native team's concerns about exposing this data, how about we remove it and expose only route-set and location-update events that are generated by the client anyway?

Reply:
We used this to debug get status polling. Specifically, would like to use it to figure out this one
https://github.com/mapbox/mapbox-navigation-android/pull/4419

I would like to keep it as it helps us figure out a lot of things.